tensorboard/plugins/debugger/tensor_helper.py

```diff
@@ -61,6 +62,54 @@ def parse_time_indices(s):
     return parsed[0]
 
 
+def translate_dtype(dtype):
+  """Translate numpy dtype into a string.
+
+  The 'object' type is understood as a tenosrflow string and translated into
+  'string'.
+
+  Args:
+    dtype: A numpy dtype object.
+
+  Returns:
+    A string representing the data type.
+  """
+  out = str(dtype)
+  # String-type TensorFlow Tensors are represented as object-type arrays in
+  # numpy. We map the type name back to 'string' for clarity.
+  return 'string' if out == 'object' else out
+
+
+def process_buffers_for_display(s, limit=40):
+  """Process a buffer for human-readable display.
+
+  This function performs the following operation on each of the buffers in `s`.
+    1. Truncate input buffer if the length of the buffer is greater than
+       `limit`.
+    2. Apply `binascii.b2a_qp` on the truncated buffer to make the buffer
+       printable and convertible to JSON.
+    3. If truncation happened (in step 1), append a string at the end
+       describing the original length and the truncation.
+
+  Args:
+    s: The buffer to be processed, either a single buffer or a nested array of
+      them.
+    limit: Length limit for each buffer, beyond which truncation will occur.
+
+  Return:
+    A single processed buffer or a nested array of processed buffers.
+  """
+  if isinstance(s, (list, tuple)):
+    return [process_buffers_for_display(elem, limit=limit) for elem in s]
+  else:
+    length = len(s)
+    if length > limit:
+      return (binascii.b2a_qp(s[:limit]) +
+              b' (length-%s truncated at %s bytes)' % (length, limit))
```

Comment:
also nit: Any reason for this to be a binary string specifically (prefixed with `b'`)?

Reply:
This is for Python2-3 compatibility. Without the b, the string will end up being a unicode string in Python3 and cannot be concatenated with the output of `binascii.b2a_qp`.